packages/commonwealth/client/scripts/views/components/tx_signing/helpers.ts

```diff
@@ -0,0 +1,118 @@
+import m from 'mithril';
+
+import { ITXModalData, TransactionStatus } from 'models';
+import { NextFn, StageName } from './types';
+
+const createProposalTransactionLabels = {
+  // substrate: accounts
+  balanceTransfer: 'Transfer balance',
+  // substrate: collective
+  createCouncilMotion: 'Create council motion',
+  voteCouncilMotions: 'Vote on council motion',
+  // substrate: elections
+  submitCandidacy: 'Submit candidacy',
+  setApprovals: 'Set election votes',
+  retractVoter: 'Retract election votes',
+  presentWinner: 'Present election winner',
+  reapInactiveVoter: 'Claim inactive voter bond',
+  // substrate: democracy
+  createDemocracyProposal: 'Create democracy proposal',
+  notePreimage: 'Note preimage',
+  noteImminentPreimage: 'Note imminnet preimage',
+  secondDemocracyProposal: 'Second democracy proposal',
+  submitDemocracyVote: 'Vote on democracy proposal',
+  submitProxyDemocracyVote: 'Vote on democracy proposal (proxy)',
+  setProxy: 'Set proxy',
+  resignProxy: 'Resign proxy',
+  removeProxy: 'Remove proxy',
+  delegate: 'Set delegate',
+  undelegate: 'Remove delegate',
+  // edgeware: treasury
+  proposeSpend: 'Propose treasury spend',
+  contractInteraction: 'Interact with contract',
+  // cosmos: accounts
+  MsgSend: 'Send balance',
+  MsgDelegate: 'Delegate stake',
+  MsgUndelegate: 'Undelegate stake',
+  MsgRedelegate: 'Redelegate stake',
+  // cosmos: governance
+  MsgDeposit: 'Increase proposal deposit',
+  MsgVote: 'Submit vote',
+  MsgSubmitProposal: 'Submit proposal',
+};
+
+export const getTransactionLabel = (txname) => {
+  return createProposalTransactionLabels[txname];
+};
+
+export const setupEventListeners = (
+  vnode: m.Vnode<
+    {
+      next: NextFn;
+    } & ITXModalData,
+    // eslint-disable-next-line @typescript-eslint/ban-types
+    { timerHandle?: NodeJS.Timeout } | {}
+  >
+) => {
+  vnode.attrs.txData.events.once(TransactionStatus.Ready.toString(), () => {
```

Comment:
This shouldn't be a general component, because it's only ever used in Substrate (and even then, likely used rarely, since our substrate support isn't great).